reflex/state.py

```diff
@@ -1277,6 +1281,17 @@ def __setattr__(self, name: str, value: Any):
             self._mark_dirty()
             return
 
+        if (
+            not name.startswith("__")
+            and not name.startswith(f"_{type(self).__name__}__")
+            and name not in self.vars
```

Comment:
nit: we may want to put this check first since it's the most common. Then if they are doing things correctly, it will avoid the above two checks